### PR TITLE
Duplicate the behaviour of `find -L` with broken symlinks

### PR DESCRIPTION
**Pull request by neersighted**
This PR mimics the behaviour of GNU `find -L -type l`; attempting to dereference symlinks while simultaneously filtering symlinks results in the list of dangling symlinks in the current tree.

While I am not entirely familiar with the implementation of symlinks on Windows, the test suite passes there as expected.

Fixes #357

### CODE REVIEW

**Comment by sharkdp** [comment]
Thank you very much for your contribution!

I didn't have to time to fully review this, but I would like us to perform some benchmarks before we move forward with this. It looks like this could be performance-critical.

**Comment by neersighted** [comment]
I don't have time for a lot of testing right now, but a quick and dirty test against the FreeBSD ports tree shows almost no performance change (with a warm cache):

```
$ hyperfine 'fd . /usr/ports' 'target/release/fd . /usr/ports'
Benchmark #1: fd . /usr/ports
  Time (mean ± σ):      1.297 s ±  0.012 s    [User: 1.673 s, System: 3.293 s]
  Range (min … max):    1.283 s …  1.315 s

Benchmark #2: target/release/fd . /usr/ports
  Time (mean ± σ):      1.306 s ±  0.012 s    [User: 1.605 s, System: 3.398 s]
  Range (min … max):    1.294 s …  1.327 s

Summary
  'fd . /usr/ports' ran
    1.01 ± 0.01 times faster than 'target/release/fd . /usr/ports'
```

Both versions of fd are built with the latest Rust nightly, in release mode.

File: src/walk.rs [hunk 1]
```diff
@@ -230,35 +237,54 @@ pub fn scan(path_vec: &[PathBuf], pattern: Arc<Regex>, config: Arc<FdOptions>) {
...
+                        path: e.into_path(),
+                    }
+                }
                 Err(err) => {
-                    tx_thread.send(WorkerResult::Error(err)).unwrap();
-                    return ignore::WalkState::Continue;
+                    // Keep track of a possible broken symlink.
+                    let mut symlink_o = None;
+                    if let ignore::Error::WithPath { ref path, ref err } = err {
+                        // An I/O error when the path does not exist (should) indicate a dangling symlink.
```
[the remark this answers is not in the document]
Comment (tavianator) [reply src/walk.rs]: Dangling symlinks are indicated by an errno of `ENOENT` or `ENOTDIR`.  `ELOOP` could be a a cycle, or just too many nested symlinks.  `EACCES` indicates a permissions error when following the link.  There are a few other errors like `EIO` and `ENOMEM` you could get as well.  Not sure what the rust analogues of all these are.

**Comment by sharkdp** [comment]
superseded by #497

**Pull request closed without merging**